### PR TITLE
[RFC] clang-format: add configuration file

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Adds a configuration file to be used with clang-format. This file was copied from the Linux kernel, but with the ForEachMacros updated for OP-TEE.

Clang-format can help with formatting the code, but we are not expecting follow it to the letter.

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
Thanks @clementfaure, that's very helpful. I'll try to update the `clang-format` file. For now, it will only as a helper.

**Comment by jforissier** [comment]
I like that, this looks immensely useful and so easy to use with vim (just bind the `=` reformatting command to `clang-format`).

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by clementfaure** [comment]
Naturally, `Acked-by: Clement Faure <clement.faure@nxp.com>`

**Comment by jenswi-linaro** [comment]
Tags applied

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,150 @@
+# SPDX-License-Identifier: GPL-2.0
+#
+# clang-format configuration file. Intended for clang-format >= 11.
+#
+# This is not perfect, but should help in most cases.
+# Please note that the maintainers has the last word if there's disagrement
```
Comment (jforissier) [inline .clang-format:6]: have

Comment (jenswi-linaro) [reply .clang-format]: Thanks, I'll fix it.

**Comment by jenswi-linaro** [comment]
Update

**Pull request merged**